dascore/proc/aggregate.py

```diff
@@ -41,12 +57,44 @@
 """
 
 
+def _get_new_coord(coord, dim_reduce):
+    """Get the new coordinate."""
+
+    def _maybe_handle_datatypes(func, data):
+        """Maybe handle the complexity of date times here."""
+        try:  # First try function directly
+            out = func(data)
+        except Exception:  # Fall back to floats and re-packing.
+            float_data = dc.to_float(data)
+            dfunc = to_datetime64 if is_datetime64(data) else to_timedelta64
+            out = dfunc(func(float_data))
+        return np.atleast_1d(out)
+
+    if dim_reduce == "empty":
+        new_coord = coord.update(shape=(1,), start=None, stop=None, data=None)
+    elif dim_reduce == "squeeze":
+        return None
+    elif (func := _AGG_FUNCS.get(dim_reduce)) or callable(dim_reduce):
+        func = dim_reduce if callable(dim_reduce) else func
+        coord_data = coord.data
+        if dtype_time_like(coord_data):
+            result = _maybe_handle_datatypes(func, coord_data)
+        else:
+            result = func(coord.data)
+        new_coord = coord.update(data=result)
+    else:
+        msg = "dim_reduce must be 'empty', 'squeeze' or valid aggregator."
+        raise ParameterError(msg)
+    return new_coord
+
+
 @patch_function()
 @compose_docstring(params=AGG_DOC_STR, options=list(_AGG_FUNCS))
 def aggregate(
     patch: PatchType,
     dim: str | Sequence[str] | None = None,
     method: str | Callable = "mean",
+    dim_reduce: str | Callable = "empty",
 ) -> PatchType:
     """
     Aggregate values along a specified dimension.
```

Comment:
The notes section in the docs should be revised based on the new `dim_reduce` argument.